### DISCUSSION TITLE
Import config type

### DISCUSSION THREAD

**Discussion by Daniel-Knights** [eta-dev/eta#221] [Q&A]
Hi, I need to use the `PartialConfig` type in my Typescript project, but can't seem to access it directly. Is there a way to do this?

My current workarounds are to either import directly from `node_modules` or use the `Parameters` utility type to extract it from the `configure` function.

**Comment by bgub** [comment]
I think the best way is like this:

```ts
import type { EtaConfig } from "eta/dist/types/config"

// and then use Partial<EtaConfig>
```

Does this work for you?

**Reply by Daniel-Knights** [reply]
TS isn't a fan: `Cannot find module 'eta/dist/types/config' or its corresponding type declarations.`

**Reply by bgub** [reply]
@Daniel-Knights strange, it's working for me. What stack are you using? And what version of `eta`?

**Reply by Daniel-Knights** [reply]
Eta: 2.0.1
TS: 5.0.2

TS config:

```json
{
  "compilerOptions": {
    "target": "ESNext",
    "module": "node16",
    "moduleResolution": "node16",
    "strict": true,
    "importHelpers": true,
    "skipLibCheck": false,
    "checkJs": true,
    "outDir": "dist",
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "sourceMap": false,
    "baseUrl": ".",
    "lib": ["ESNext", "DOM"],
    "paths": {
      "@compodoc/live-server": ["lib/live-server.d.ts"]
    }
  },
  "include": ["lib/**/*.ts"],
  "exclude": ["node_modules"]
}
```

Could be wrong, but I don't think it should work with TS as `'eta/dist/types/config'` isn't defined in `package.json` `exports`

**Reply by gurgunday** [reply]
Yeah, I think package.json definitions are required for "moduleResolution": "node16" and "moduleResolution": "nodenext". As a workaround, you can try switching it to node. I'll raise the issue.
